### ISSUE TITLE
Data binding not working for ion-input with type date or time

### ISSUE THREAD

**Issue by rveerd** [ionic-team/ionic#15285]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
√ Gathering environment info - done!

Ionic:

   ionic (Ionic CLI)          : 4.1.0 (D:\npm\node_modules\ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.2
   @angular-devkit/core       : 0.7.3
   @angular-devkit/schematics : 0.7.3
   @angular/cli               : 6.1.3
   @ionic/ng-toolkit          : 1.0.6
   @ionic/schematics-angular  : 1.0.5

Cordova:

   cordova (Cordova CLI) : not installed
   Cordova Platforms     : not available
   Cordova Plugins       : not available

System:

   Android SDK Tools : 26.1.1
   NodeJS            : v8.11.3 (C:\Program Files\nodejs\node.exe)
   npm               : 5.6.0
   OS                : Windows 7

Environment:

   ANDROID_HOME : D:\Android\Sdk
```

**Describe the Bug**
Two-way data binding is not working with `<ion-input>` with `type="date"` or `type="time"`. If you change a value in the page, the value in the controller is not updated.

**Steps to Reproduce**
1. Create a component with a `<ion-input>` of `type` `"date"` or `"time"` and two-way data binding (see code below).
2. Open the page and change the value.
3. The value in the controller is not updated.
4. If you change the `<ion-input>` to `<input>` OR change the `type` to `"text"` it does work and the value in the controller is updated.

**Related Code**

```
<ion-input type="time" [(ngModel)]="item.when" name="when"></ion-input>
```

**Expected Behavior**
I expect value in the controller to change when I use two-way data binding.

**Additional Context**
n/a

**Comment by mhartington** [comment]
Currently there's no `control-value-accessor` for Date/Time inputs. Will put this on the back log for now.

**Comment by NiklasMerz** [comment]
I have a similiar problem with the change event for inputs with type file:
`<ion-input type="file"  (change)="onFileChange($event)"></ion-input>`.

Ion-input does not call the change function in Chrome. Firefox works. A normal input does work in Chrome.

**Comment by paulstelzer** [comment]
I close this for now because it should fixed with #17043 . After rc.2 release you can check again and if it still not working, just create a comment here and I reopen

I also check input type file and change and this is working with this fix, too
